### PR TITLE
python3 fixes

### PR DESCRIPTION
**Pull request by thomasgubler**
Let's see what python2/travis does with these changes...

### CODE REVIEW

**Comment by LorenzMeier** [comment]
@thomasgubler It doesn't like it yet.

**Comment by thomasgubler** [comment]
Note to myself for later:
```
  File "/home/travis/build/PX4/Firmware/Tools/px_romfs_pruner.py", line 84, in <module>
    main()
  File "/home/travis/build/PX4/Firmware/Tools/px_romfs_pruner.py", line 80, in main
    f.write(bytes(pruned_content, "UTF-8"))
TypeError: str() takes at most 1 argument (2 given)
make[3]: *** [src/firmware/nuttx/romfs.o] Error 1
make[2]: *** [src/firmware/nuttx/CMakeFiles/romfs.dir/all] Error 2
make[2]: *** Waiting for unfinished jobs....
```

**Comment by thomasgubler** [comment]
rebased and improved, even travis is happy now

File: Tools/px_romfs_pruner.py [hunk 1]
```diff
@@ -77,7 +77,7 @@ def main():
 
                         # overwrite old scratch file
                         with open(file_path, "wb") as f:
-                                f.write(pruned_content)
+                                f.write(pruned_content.encode("UTF-8"))
```
Comment (LorenzMeier) [inline Tools/px_romfs_pruner.py]: I'm not so sure about UTF-8. NuttX can't read that. Could you please choose a plain ASCII encoding like latin?

**Comment by LorenzMeier** [comment]
Have you flashed and tested this? The UTF8 encoding could easily result in a boot failure.

**Comment by thomasgubler** [comment]
@LorenzMeier Ups thanks, I looked at the wrong doc for the default of the python call I replaced. Since we only have ascii chars in our scripts the previous version did indeed work however. I have now changed the encode call to a make sure it gives an error when a non ascii char is encountered.

**Comment by LorenzMeier** [comment]
Thanks!

**Pull request merged**